Add autoconfigure support for OTLP gzip compression (#3589)
open-telemetry/opentelemetry-java: sdk-extensions/autoconfigure/README.md

| otel.exporter.otlp.headers   | OTEL_EXPORTER_OTLP_HEADERS  | Key-value pairs separated by commas to pass as request headers on OTLP trace and metrics requests.        |
| otel.exporter.otlp.traces.headers   | OTEL_EXPORTER_OTLP_TRACES_HEADERS  | Key-value pairs separated by commas to pass as request headers on OTLP trace requests.        |
| otel.exporter.otlp.metrics.headers   | OTEL_EXPORTER_OTLP_METRICS_HEADERS  | Key-value pairs separated by commas to pass as request headers on OTLP metrics requests.        |

> | otel.exporter.otlp.compression   | OTEL_EXPORTER_OTLP_COMPRESSION | The key of the compression type to use on OTLP trace and metric requests. Options include `gzip`. By default no value is specified. |

Reviewer: ```suggestion
| otel.exporter.otlp.compression   | OTEL_EXPORTER_OTLP_COMPRESSION | The compression type to use on OTLP trace and metric requests. Options include `gzip`. By default no compression will be used. |
```

Reviewer: ditto changes on the other 2 lines below
